src/CarbonSystemSolvers.jl

```diff
@@ -32,139 +32,149 @@ struct CarbonSolverApprox
 
 TBW
 """
-function CarbonSolverApprox(
-        Θᶜ      :: Float64 = 25.0,
-        Sᴬ      :: Float64 = 35.0,
-        Δpᵦₐᵣ   :: Float64 = 0.0,
-        Cᵀ      :: Float64 = 2050.0e-6,
-        Aᵀ      :: Float64 = 2350.0e-6,
-        pH      :: Float64 = 8.0,
-        pCO₂ᵃᵗᵐ :: Float64 = 280.0e-6,
-        )
+@inline function CarbonSystemApprox(
+    Θᶜ      :: FT = 25,
+    Sᴬ      :: FT = 35,
+    Δpᵦₐᵣ   :: FT = 0,
+    Cᵀ      :: FT = 2050e-6,
+    Aᵀ      :: FT = 2350e-6,
+    pH      :: FT = 8,
+    pCO₂ᵃᵗᵐ :: FT = 280e-6,
+    ) where {FT}
```

Comment:
Note that this syntax "diagonalizes" the input, in that all input parameters are required to be the same type `FT`.

This could be desired or not --- if its user-facing, then such type restrictions are "unfriendly", because it disallows things like `a = 1, b = 2.1`, requiring the user to write `a=1.0` (for no good reason in that case).

Internally you may want to diagonalize types for safety reasons; ie you know that the inputs should always be the same type and you want to catch a bug. We don't usually use that pattern because it goes against YASGuide but can be warranted for certain code, especially very complex / low-level code.

I think this function signature may be invalid, however --- some of the defaults are `Int` and some are `Float64`, but the annotations `::FT` are requiring all inputs to be the same type.

Reply:
Ok, we should discuss the use of `FT` a bit more. 

I did find that this particular signature was problematic and led to some really weird behaviour. Changing the ints back to floats fixed it, though. The intention is that these functions should be fed with model fields.